test/test_readonly_task.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "syskit/test/self"
+
+module Syskit
+    describe TaskContext do
+        before do
+            task_m = TaskContext.new_submodel do
+                property "p", "/double"
+            end
+
+            @task = syskit_stub_deploy_and_configure(task_m.with_arguments(read_only: true))
+            Orocos.allow_blocking_calls { @task.orocos_task.configure(false) }
+        end
+
+        after do
+            state = Orocos.allow_blocking_calls { @task.orocos_task.rtt_state }
+            if state == :RUNNING
+                Orocos.allow_blocking_calls { @task.orocos_task.stop(false) }
+            end
+
+            execute { @task.stop! } if @task.started?
+        end
+
+        describe "read-only tasks" do
+            it "raises when attempting to change a property" do
+                task = @task
+
+                assert_raises(InvalidReadOnlyOperation) do
+                    task.properties.p = 2.0
+                end
+            end
+
+            it "emits start when it is started while the component is running" do
+                task = @task
+                Orocos.allow_blocking_calls { task.orocos_task.start(false) }
+
+                expect_execution { task.start! }
+                    .to { emit task.start_event }
+            end
+
+            it "does not emit start if started while the component is not running" do
+                task = @task
+
+                expect_execution do
+                    Orocos.allow_blocking_calls { task.orocos_task.start(false) }
+                end.to { not_emit task.start_event }
+            end
+
+            it "emits start when the component is running and the task is starting" do
+                task = @task
+                execute { task.start! }
+
+                assert task.starting?
+
+                expect_execution do
+                    Orocos.allow_blocking_calls { task.orocos_task.start(false) }
+                end.to { emit task.start_event }
+            end
+
+            it "does not emit interrupted if the task is stopped" do
+                task = @task
+                Orocos.allow_blocking_calls { task.orocos_task.start(false) }
+                execute { task.start! }
+
+                expect_execution { task.stop! }
+                    .to { not_emit task.interrupt_event }
+            end
```

Comment:
Need a test that checks what happens when the task is running but the component has been stopped.

Reply:
(I think it should just emit stop)

Reply:
done